sklearn/preprocessing/_encoders.py

```diff
@@ -372,47 +372,30 @@ def _handle_deprecations(self, X):
                     )
                     warnings.warn(msg, DeprecationWarning)
             else:
-
                 # check if we have integer or categorical input
                 try:
                     check_array(X, dtype=np.int)
                 except ValueError:
                     self._legacy_mode = False
                     self._categories = 'auto'
                 else:
-                    if self.drop is None:
-                        msg = (
-                            "The handling of integer data will change in "
-                            "version 0.22. Currently, the categories are "
-                            "determined based on the range "
-                            "[0, max(values)], while in the future they "
-                            "will be determined based on the unique "
-                            "values.\nIf you want the future behaviour "
-                            "and silence this warning, you can specify "
-                            "\"categories='auto'\".\n"
-                            "In case you used a LabelEncoder before this "
-                            "OneHotEncoder to convert the categories to "
-                            "integers, then you can now use the "
-                            "OneHotEncoder directly."
-                        )
-                        warnings.warn(msg, FutureWarning)
-                        self._legacy_mode = True
-                        self._n_values = 'auto'
-                    else:
-                        msg = (
-                            "The handling of integer data will change in "
-                            "version 0.22. Currently, the categories are "
-                            "determined based on the range "
-                            "[0, max(values)], while in the future they "
-                            "will be determined based on the unique "
-                            "values.\n The old behavior is not compatible "
-                            "with the `drop` paramenter. Instead, you "
-                            "must manually specify \"categories='auto'\" "
-                            "if you wish to use the `drop` parameter on "
-                            "an array of entirely integer data. This will "
-                            "enable the future behavior."
-                        )
-                        raise ValueError(msg)
```

Comment:
@drewmjohnston what is the reason you removed this ValueError again (for the case where we would normally end up in the legacy behaviour, but where the user did specify the `drop` keyword).

Reply:
Just tried this out locally. So you still get an error, but the one related to `n_values` or `categorical_features` not being compatible with `drop`. Which is a bit confusing here, since you didn't specify those. So I would either add it back, or change the other error message a bit.